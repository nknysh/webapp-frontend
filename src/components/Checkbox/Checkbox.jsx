import React from 'react';
import { FormControlLabel } from '@material-ui/core';

import { propTypes, defaultProps } from './Checkbox.props';
import { StyledCheckBox, MaterialCheckbox } from './Checkbox.styles';

export const Checkbox = ({ label, onSelected, ...props }) => {
  const onChange = e => onSelected(e.currentTarget.checked);

  return (
    <StyledCheckBox>
      <FormControlLabel control={<MaterialCheckbox onChange={onChange} {...props} />} label={label} />
    </StyledCheckBox>
  );
};

Checkbox.propTypes = propTypes;
Checkbox.defaultProps = defaultProps;

export default Checkbox;
