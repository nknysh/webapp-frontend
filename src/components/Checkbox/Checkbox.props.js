import PropTypes from 'prop-types';

import { noop } from 'utils';

export const propTypes = {
  label: PropTypes.string,
  onSelected: PropTypes.func,
};

export const defaultProps = {
  onSelected: noop,
};
