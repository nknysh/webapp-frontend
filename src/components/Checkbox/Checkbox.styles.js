import styled from 'styled-components';
import { Checkbox } from '@material-ui/core';

import theme from 'styles/theme';

export const StyledCheckBox = styled.div`
  label,
  label span {
    font-family: ${theme.defaultFont} !important;
  }
`;

export const MaterialCheckbox = styled(Checkbox)`
  svg {
    fill: ${theme.primary};
  }
`;
