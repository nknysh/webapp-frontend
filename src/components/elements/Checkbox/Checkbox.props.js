import PropTypes from 'prop-types';

export const propTypes = {
  label: PropTypes.any,
};

export const defaultProps = {};
