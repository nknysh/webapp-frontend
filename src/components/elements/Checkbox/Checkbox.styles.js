import styled from 'styled-components';
import { Checkbox } from '@material-ui/core';

import { theme } from 'styles';

export const StyledCheckBox = styled.div`
  label,
  label span {
    font-family: ${theme.fonts.defaultFont} !important;
  }
`;

export const MaterialCheckbox = styled(Checkbox)`
  svg {
    fill: ${theme.primary};
  }
`;
