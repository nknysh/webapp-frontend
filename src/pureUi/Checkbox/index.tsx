import React, { EventHandler, FormEvent } from 'react';
import styled from 'styled-components';
import { pureUiTheme } from 'pureUi/pureUiTheme';

export interface CheckboxProps extends React.HTMLProps<HTMLInputElement> {}

const Checkbox = (props: CheckboxProps) => {
  const { className, ...checkboxProps } = props;
  return (
    <span className={className}>
      <input type="checkbox" {...checkboxProps} />
      <span className="surrogate" />
    </span>
  );
};

export default styled(Checkbox)`
  position: relative;
  width: ${pureUiTheme.measurements.checkboxSize}px;
  height: ${pureUiTheme.measurements.checkboxSize}px;
  display: inline-block;
  vertical-align: middle;
  margin: 1px;

  & > input {
    position: absolute;
    appearance: none;
    width: ${pureUiTheme.measurements.checkboxSize}px;
    height: ${pureUiTheme.measurements.checkboxSize}px;
    opacity: 0;
    margin: 0;
    padding: 0;
    cursor: pointer;
    z-index: 1;

    &:disabled {
      cursor: default;
    }
  }

  .surrogate {
    transition: all 0.15s ease-out;
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    z-index: 0;
    border: ${pureUiTheme.colorRoles.lightGreyBorder} 1px solid;
    background-color: ${pureUiTheme.colors.white};
    box-shadow: 0 0 0 5px transparent;
  }

  & > input:checked ~ .surrogate {
    border: ${pureUiTheme.colors.gold} 1px solid;
    background-color: ${pureUiTheme.colors.gold};
  }

  & > input:focus ~ .surrogate,
  & > input:active ~ .surrogate {
    box-shadow: 0 0 0 2px ${pureUiTheme.colors.marine};
  }

  & > input ~ .surrogate:after {
    content: '';
    display: block;
    position: absolute;
    transition: all 0.15s ease-out;
    background-color: transparent;
    top: 1px;
    left: 1px;
    width: ${pureUiTheme.measurements.checkboxSize - 2}px;
    height: ${pureUiTheme.measurements.checkboxSize - 2}px;
  }

  & > input:checked ~ .surrogate:after {
    border-radius: ${pureUiTheme.measurements.checkboxTittleSize}px;
    top: ${pureUiTheme.measurements.controlDottOffset}px;
    left: ${pureUiTheme.measurements.controlDottOffset}px;
    width: ${pureUiTheme.measurements.checkboxTittleSize}px;
    height: ${pureUiTheme.measurements.checkboxTittleSize}px;
    background-color: ${pureUiTheme.colors.white};
  }
`;
