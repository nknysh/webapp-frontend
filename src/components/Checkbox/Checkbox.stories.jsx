import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';

import Checkbox from './Checkbox';

storiesOf('Checkbox', module)
  .addDecorator(withKnobs)
  .add('default', () => {
    return <Checkbox />;
  });
