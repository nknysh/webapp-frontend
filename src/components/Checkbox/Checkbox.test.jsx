import React from 'react';

import { Checkbox } from './Checkbox';

const getComponent = props => shallow(<Checkbox {...props} />);

describe('<Checkbox />', () => {
  describe('render', () => {
    it('matches snapshot', () => {
      expect(getComponent()).toMatchSnapshot();
    });
  });
});
