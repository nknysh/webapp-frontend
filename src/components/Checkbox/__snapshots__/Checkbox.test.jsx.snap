// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> render matches snapshot 1`] = `
<styled.div>
  <WithStyles(WithFormControlContext(FormControlLabel))
    control={<ForwardRef />}
  />
</styled.div>
`;
